src/librustc_metadata/encoder.rs

```diff
@@ -839,7 +839,9 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
                 _ => None,
             },
             mir: match item.node {
-                hir::ItemStatic(..) |
+                hir::ItemStatic(..) if self.tcx.sess.opts.debugging_opts.always_encode_mir => {
```

Comment:
Indeed it was, there were a few things in the PR introducing this that are less than ideal. They skipped review because they weren't in the original PR state but rather in "fixes for tests".